endpoints/openrtb2/auction.go

```diff
@@ -214,8 +205,31 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 
 	secGPC := r.Header.Get("Sec-GPC")
 
+	var resolvedBidReq json.RawMessage
+	requestExt, _ := req.GetRequestExt()
+	if req.Test == 1 || (requestExt != nil && requestExt.GetPrebid() != nil && requestExt.GetPrebid().Debug) {
```

Comment:
I tried two approaches that we discussed and didn't like any of them.
1) If I move FPD extraction down to `HoldAuction` - it will be executed for amp and video endpoints too, and we don't want this. 
2) If I move `getDebugInfo` up to `action` endpoint handler - I will need to duplicate it for amp and video as well. 

Line 210 is a duplicate of `parseRequestDebugValues` function. For resolved request we only need to check `req.Test` is 1 or `req.ext.Prebid.debug` is enabled. For action endpoint DebugLog is always nil, so this simple check does the thing.
This approach is some sort of cheating, I understand. I didn't find another good way to implement this. 
We discussed it with team and they are ok with this approach. 
Please let me know what you think.

Reply:
> it will be executed for amp and video endpoints too, and we don't want this.

We actually do want to run FPD logic for AMP, per https://github.com/prebid/prebid-server/issues/1727. The HoldAuction endpoint also has knowledge of which endpoint it is serving through AuctionRequest.RequestType.

Reply:
FPD support fr AMP endpoint is not supported in PBS-go. In case we will need to do this, then the only way to do this - is to move FPD extraction inside of HoldAction. I will update the code this way.

Reply:
Code updated, please re-review